### ISSUE TITLE
Double new lines don't end up in a paragraph

### ISSUE THREAD

**Issue by MightyCreak** [vector-im/riotX-android#1237]
**Problem**
When I write a message on RiotX and use double new lines in my message, I don't get paragraphs in the posted text.

**Actual result**
Just one new line.

**Expected result**
I am expecting to see a small space between the paragraphs.

**Repro steps**
* Write this message:

      Paragraph 1
      
      Paragraph 2

* Send it
* See the rendered/posted text, it looks like this:

      Paragraph 1
      Paragraph 2

**Comment by tulir** [comment]
@bmarty isn't this more of a HTML rendering thing rather than composer? When you enter two newlines, it *does* end up in a paragraph (you can find the `<p>` tags in view source), but RiotX renders paragraphs as just a newline

**Comment by MightyCreak** [comment]
@tulir it is indeed, I verified on Riot desktop and the text I posted with RiotX does have space between paragraphs.

**Comment by vberger** [comment]
Looks like it's a little more subtle actually. I just made a few tests and, given this input text in riotX:

```
First line.

Second line.
```

The resulting JSON is:

```json
{
    "msgtype": "m.text",
    "body": "First line.\nSecond line.",
    "format": "org.matrix.custom.html",
    "formatted_body": "<p>First line.</p><p>Second line.</p>"
  }
```

So, the formatted body properly contains two paragraphs, that riotX fails to display as separated with a blank line, but riot-web does display correctly. However, we can see that the raw body now contains a single newline, as opposed to the two I typed in.

Interestingly, if I remove the punctuation in the message, it is now processed into this JSON

```json
{
    "msgtype": "m.text",
    "body": "First line\n\nSecond line"
}
```

Keeping the two newlines and displaying it as expected everywhere.

----

If I ensure that the message is markdown-processed by adding formatting, I can compare riotX & riot-web outputs:

```
First *line*.

Second **line**.
```

RiotX turns this message into this JSON:

```json
{
    "msgtype": "m.text",
    "body": "First line.\nSecond line.",
    "format": "org.matrix.custom.html",
    "formatted_body": "<p>First <em>line</em>.</p><p>Second <strong>line</strong>.</p>"
}
```

while riot-web turns it into this:

```json
{
    "msgtype": "m.text",
    "body": "First *line*.\n\nSecond **line**.",
    "format": "org.matrix.custom.html",
    "formatted_body": "<p>First <em>line</em>.</p>\n<p>Second <strong>line</strong>.</p>\n"
}
```

riotX seems to eat the multiple newlines when it markdown-processes the message, while riot-web does not, in addition to the displaying issue.

**Comment by dkasak** [comment]
This is turning up into a pretty big usability nightmare, since users on Element Android are expecting that their input Markdown syntax was wrong because they see their messages rendered incorrectly. They then attempt to edit the message in order to correct it only to garble the message for everyone else.

Could we perhaps prioritize the fix for this? Correct message rendering is pretty essential for a nice user experience since that's the feature users are in contact with for the largest amount of time when using Element.

**Comment by MightyCreak** [comment]
I completely agree. At first it's just not great, but as soon as your start editing the message, it's becoming worse and worse.

**Comment by MightyCreak** [comment]
Since the bug is one year old, I've tried again @vberger's tests to reproduce the issue.

So I entered this message in Element-Android:
```
Line one.

Line two.
```

I enabled dev mode and looked at the decrypted source:
```json
{
  "type": "m.room.message",
  "content": {
    "msgtype": "m.text",
    "body": "Line one.

Line two.",
    "format": "org.matrix.custom.html",
    "formatted_body": "<p>Line one.</p>
<p>Line two.</p>"
  },
  "room_id": "!redacted:matrix.org"
}
```
Compared to @vberger, the `formatted_body` includes the newline now (newlines are displayed as-is instead of being displayed as `\n`, I respected this when I handwrote the text here).

The rendered result on Element-Android:
![image](https://user-images.githubusercontent.com/6382548/114309519-69d11400-9ab5-11eb-8b53-aecac324cf02.png)

The rendered result on Element-Desktop:
![image](https://user-images.githubusercontent.com/6382548/114309498-4f973600-9ab5-11eb-8e23-ef82e7649e0a.png)

We can easily see that Element-Desktop respects the space between the two paragraphs.
______________________________________________

Now, if I enter this message (note the missing punctuation):
```
Line one

Line two
```

The decrypted source is like so:
```json
{
  "type": "m.room.message",
  "content": {
    "msgtype": "m.text",
    "body": "Line one

Line two"
  },
  "room_id": "!redacted:matrix.org"
}
```
As with @vberger, the `format` and `formatted_body` fields are missing. And it has an impact on the rendered result:

* Element-Android:  
  ![image](https://user-images.githubusercontent.com/6382548/114310690-9edf6580-9ab9-11eb-8be3-ebf27c44fdd2.png)
* Element-Desktop:  
  ![image](https://user-images.githubusercontent.com/6382548/114310712-b1f23580-9ab9-11eb-95fa-c8d14013c1fe.png)

So here, markdown is not respected (I think) and paragraphs are not created in either devices.
______________________________________________

Last test, with the markdown formatting. I entered this text:
```
Line *one*.

Line **two**.
```
Here's the decrypted source:
```json
{
  "type": "m.room.message",
  "content": {
    "msgtype": "m.text",
    "body": "Line *one*.

Line **two**.",
    "format": "org.matrix.custom.html",
    "formatted_body": "<p>Line <em>one</em>.</p>
<p>Line <strong>two</strong>.</p>"
  },
  "room_id": "!redacted:matrix.org"
}
```
The decrypted source is the same on Element-Desktop, except it show `\n` instead of newlines, but I guess it's just a difference in the way to render the JSON source.

The render is good in both devices (except the same missing inter-paragraph space):
* Element-Android:  
  ![image](https://user-images.githubusercontent.com/6382548/114310971-d864a080-9aba-11eb-9a9f-d7519e85ddee.png)
* Element-Desktop:
  ![image](https://user-images.githubusercontent.com/6382548/114310997-ec100700-9aba-11eb-933f-f07d6ae505a6.png)
_____________________________________

Some information on my devices:
* Element-Andoid: 1.1.3, PlayStore
* Element-Desktop: 1.7.24, Flatpak on Fedora 33
